common/src/services/environment.service.ts

```diff
@@ -90,22 +171,24 @@ export class EnvironmentService implements EnvironmentServiceAbstraction {
         this.eventsUrl = urls.events;
         this.enterpriseUrl = urls.enterprise;
 
-        const envUrls = new EnvironmentUrls();
-        if (this.baseUrl) {
-            envUrls.base = this.baseUrl;
-        } else {
-            envUrls.api = this.apiUrl;
-            envUrls.identity = this.identityUrl;
-            envUrls.events = this.eventsUrl;
-        }
+        // TODO: Emit event to notify other services they should change urls?
```

Comment:
I really do like this idea (see comments in response to your earlier comment) now that I fully understand the scope of the changes and impact/needs around when those URLs change (through settings after logging out).

Reply:
I might be slow in understanding this, but don't all the other services just grab the URL from this one every time?

Reply:
They do, but `NotificationService` initializes a SignalR session, which would still call the old endpoint. Until it's been re-initialized hence the call on https://github.com/bitwarden/jslib/pull/435/files#diff-486e7d088c99f1fb43091643e50264a6272c5d14e3f572cfb04deb845130ded9R50